### PR TITLE
Convert volumes object into array of volumes

### PR DESCRIPTION
**Pull request by gregarndt**
This is to have a volumes array rather than an object keyed based on the
volume type.

### CODE REVIEW

File: plugins/volume/volumeset.go [hunk 1]
```diff
@@ -38,28 +51,24 @@ type volumeSet struct {
...
 				"Unrecognized volume type. This should never happen and " +
 					"is considered a fatal error.",
 			)
 		}
+		opts := &volumeOptions{
+			taskID: context.TaskInfo.TaskID,
+			runID:  context.TaskInfo.RunID,
+			spec:   spec(volume),
+		}
+		v, err := plugin.newVolume(opts)
```
Comment (gregarndt) [inline plugins/volume/volumeset.go]: maybe implement opts as a "map[string]interface{}" and then the plugin will do what it needs to with the options.

**Comment by petemoore** [comment]
@gregarndt is this PR ready for review?

**Comment by gregarndt** [comment]
I think this is going to be closed.  @jonasfj is rewriting it.

**Comment by jonasfj** [comment]
Yes, let's closet this and rethink volumes and how they take options.

**Pull request closed without merging**